In [5]:
from pyvis.network import Network
import networkx as nx
import numpy as np
import scipy
from matplotlib import pyplot as plt 

## Read Gene-Cell

Data Source: https://www.cell.com/cell/fulltext/S0092-8674(18)30955-3?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867418309553%3Fshowall%3Dtrue

In [2]:
from scipy import sparse, io

RNA_seq_matrix = io.mmread("data/FC_data_LRcell.mm.txt")

In [3]:
print("RNA_seq_matrix.shape: ", RNA_seq_matrix.shape)
print("Non-zero entry count=%d (%.2f%%)" % (RNA_seq_matrix.getnnz(), 
      RNA_seq_matrix.getnnz()/RNA_seq_matrix.shape[0]/RNA_seq_matrix.shape[1]*100))
print("Entry analysis: min=%d, max=%d, mean=%.3f" % (RNA_seq_matrix.min(), 
                                                     RNA_seq_matrix.max(),
                                                     RNA_seq_matrix.mean(),
                                                    ))

RNA_seq_matrix.shape:  (15976, 71639)
Non-zero entry count=137124604 (11.98%)
Entry analysis: min=0, max=1711, mean=0.251


RNA_seq_matrix is a sparse matrix where each entry stores gene expression "value" for specific gene (row) and cell (column).
The gene expression ranges from 0 to 1711, and the value is integer. 
Moreover, the data is reletively large scale and may contain noisy data, hence it is necessary to sample a subset for further analysis.

In [8]:
# cell down-sampling
cell_vec = RNA_seq_matrix.sum(0)
print("cell_vec shape =", cell_vec.shape)
print("Entry analysis: min=%d, max=%d, mean=%.3f, std=%.3f" % (cell_vec.min(), cell_vec.max(), 
                                                               cell_vec.mean(), cell_vec.std()))

cell_vec shape = (1, 71639)
Entry analysis: min=611, max=73424, mean=4009.060, std=3355.501


In [ ]:
# gene down-sampling